#118: Deconstruct Normandy

andymikulski opened the pull request:
- Fixes #118
- Removes `normandy` folder, merges most `console` folders and files together.
- Removes `NormandyLink` and custom router business
- Merges Normandy's routes in with the `console`'s router
- Removes app-specific startup from `config-overrides.js`

To review, I suggest pulling the code down and checking out the new structure. Most of the code changes are simply renaming imports.

magopian commented on src/console/App.js:
```diff
@@ -3,12 +3,12 @@ import PropTypes from 'prop-types';
 
-import Normandy from 'normandy/App';
-import { BrowserRouter, NavLink, Link } from 'react-router-dom';
+// import Normandy from 'console/App';
```
Do you want to remove that line?

magopian commented on src/console/App.test.js:
```diff
@@ -1,5 +1,5 @@
 import App from './App';
-import { wrapMockStore } from 'normandy/tests/mockStore';
+import { wrapMockStore } from 'console/tests/mockStore';
```
I find that a bit confusing: importing `console/tests/` (which is aliased to `src/tests/`) from `src/console/`

andymikulski replied:
The `console` subfolder has been removed, so now we can trust that `console/...` is pointing to the root of the source files.

magopian commented on src/components/App.js:
```diff
@@ -0,0 +1,40 @@
+import { Layout } from 'antd';
```
Is this file used somewhere? I can't see it imported anywhere

andymikulski replied:
On line 9 the `Content` and `Header` components are abstracted from `Layout`

magopian replied:
I was talking about the file itself, the `src/components/App.js` file, i don't see it imported/used anywhere, I must be missing something...

andymikulski replied:
`import DevConsoleApp from 'console/components/App';`, line 3 of `src/index.js`

magopian replied:
Right, I'm just blind. Sorry about the noise.

andymikulski commented:
When I was working on it I had a good reason to keep the `console` folder, but I can't remember what it was. I've broken down the folder as expected and consolidated the various `App.js` files.
Also, I've moved the `selectors`/`reducers` file from the `console` dir and moved them to the new dir `console/state/auth`.

magopian commented:
I think this is ready to merge, what do you think @andymikulski ?
